extensions/git/src/commands.ts

```diff
@@ -1251,7 +1251,7 @@ export class CommandCenter {
 
 		if (promptToSaveFilesBeforeCommit !== 'never') {
 			let documents = workspace.textDocuments
-				.filter(d => !d.isUntitled && d.isDirty && isDescendant(repository.root, d.uri.fsPath));
+				.filter(d => !d.isUntitled && d.uri.scheme === 'file' && isDescendant(repository.root, d.uri.fsPath));
```

Comment:
No, I think the fix should rather be around line 1271... all files should be staged, not just the list of `documents`.

Reply:
But having `d.isDirty` is excluding the documents that are saved. That's why I removed it.

Reply:
Calling `add([])` will stage everything.